### PR TITLE
[#941] Modified all BasePlugin's to inherit from a shared base class

### PR DESCRIPTION
**Pull request by Bartvaderkin**

### CODE REVIEW

File: src/openforms/appointments/base.py [hunk 1]
```diff
@@ -72,12 +72,11 @@ def __str__(self):
         return self.identifier
 
 
-class BasePlugin:
+class BasePlugin(AbstractPlugin):
```
Comment (Bartvaderkin) [inline src/openforms/appointments/base.py:75]: naming suggestions welcome :shrug:

File: src/openforms/plugins/plugin.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+from typing import List, Tuple
+
+from django.utils.translation import gettext_lazy as _
+
+
+class AbstractPlugin:
```
Comment (Bartvaderkin) [inline src/openforms/plugins/plugin.py:6]: naming suggestions welcome :shrug:

Comment (sergei-maertens) [reply src/openforms/plugins/plugin.py:6]: `AbstractBasePlugin` is fine, similar to Django's `AbstractBaseUser` class

File: src/openforms/appointments/base.py [hunk 3]
```diff
@@ -4,14 +4,14 @@
...
 from django.urls import reverse
-from django.utils.translation import gettext_lazy as _
 
 from rest_framework import serializers
 
 from openforms.submissions.models import Submission
 from openforms.utils.mixins import JsonSchemaSerializerMixin
 from openforms.utils.urls import build_absolute_uri
 
+from ..plugins.plugin import AbstractPlugin
```
Comment (sergei-maertens) [inline src/openforms/appointments/base.py:14]: please use absolute imports when importing outside of the django app/openforms module

File: src/openforms/payments/base.py [hunk 4]
```diff
@@ -2,13 +2,13 @@
...
 
 from django.http import HttpRequest, HttpResponse
-from django.utils.translation import gettext_lazy as _
 
 from rest_framework import serializers
 from rest_framework.reverse import reverse
 
 from openforms.utils.mixins import JsonSchemaSerializerMixin
 
+from ..plugins.plugin import AbstractPlugin
```
Comment (sergei-maertens) [inline src/openforms/payments/base.py:11]: please use absolute imports when importing outside of the django app/openforms module

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**